attentionbench/problems.py

```diff
@@ -1,62 +1,43 @@
-    datatypes : list[str]):
-
-    for B, H, S_Q, S_KV, DH, datatype in itertools.product(batch_sizes, head_counts, seq_lengths, seq_lengths, head_dims, datatypes):
-        bytes = B * H * 2 * (2 * S_KV * DH + 2 * S_Q * DH + S_Q * S_KV)
-        if bytes < 1e9:
-            configs.append((name, B, H, S_Q, S_KV, DH, datatype))
-
-def llama70battention(configs):
-    generate_attention_shapes(
-        configs,
-        "llama70battention",
-        batch_sizes=[1, 2, 4],
-        head_counts=[32, 40, 64],
-        head_dims=[128],
-        seq_lengths=[1024, 2048, 4096],
-        datatypes=["f16"],
-    )
-
-def sdxlattention(configs):
-    for B, H, S_Q, S_KV, DH, _ in SDXL_ATTN:
-        bytes = B * H * 2 * (2 * S_KV * DH + 2 * S_Q * DH + S_Q * S_KV)
-        if bytes < 1e9:
-            for datatype in ["f16"]:
-                configs.append(("sdxlattention", B, H, S_Q, S_KV, DH, datatype))
-
-def flash_attention(configs):
-    # batch_sizes = [1, 2, 4, 8, 16, 32]
-    # head_counts = [12, 24, 36, 42, 48]
-    # head_dims = [32, 64, 128]
-    # seq_lengths = [64, 128, 256, 384, 512, 1024, 2048, 4096, 8192, 16384, 32768, 64320]
-    # datatypes = ["f16"]
-
-    # shapes = []
-    # shapes = [
-    #     (1, 42, 384, 64320, 64, "f16"),
-    #     (1, 42, 4096, 4096, 64, "f16"),
-    #     (1, 42, 384, 4096, 64, "f16"),
-    #     (1, 42, 8192, 8192, 64, "f16"),
-    #     (1, 42, 384, 8192, 64, "f16"),
-    #     (1, 42, 16384, 16384, 64, "f16"),
-    #     (1, 42, 384, 16384, 64, "f16"),
-    # ]
-    
-    # yield from generate_attention_shapes("generalattention", batch_sizes, head_counts, head_dims, seq_lengths, datatypes)
-    llama70battention(configs)
-    sdxlattention(configs)
+from attention_utils import AttentionConfig
+
+
+def llm_sweep(configs: list[AttentionConfig], dtype: str):
+    # Batch sweep (batch * num_heads)
+    for B in [1, 2, 4, 8, 16, 32, 48, 64, 96, 128, 192]:
+        # M, K2 sweep (seq length)
+        for M in [1024, 2048, 4096, 8192, 16384]:
+            K2 = M
+            # K1, N sweep (head dim)
+            for N in [64, 128]:
+                K1 = N
+                configs.append(AttentionConfig(B, M, N, K1, K2, dtype))
+
+
+def sdxl_unet_sweep(configs: list[AttentionConfig], dtype: str):
+    sdxl_attn_shapes = [
+        (20, 4096, 64, 64, 4096),
+        (20, 4096, 64, 64, 64),
+        (40, 1024, 64, 64, 1024),
+        (40, 1024, 64, 64, 64),
+    ]
+    for B, M, N, K1, K2 in sdxl_attn_shapes:
+        configs.append(AttentionConfig(B, M, N, K1, K2, dtype))
+
+
+def bert_sweep(configs: list[AttentionConfig], dtype: str):
+    sdxl_attn_shapes = [
+        (20, 4096, 64, 64, 4096),
+        (20, 4096, 64, 64, 64),
+        (40, 1024, 64, 64, 1024),
+        (40, 1024, 64, 64, 64),
+    ]
+    for B, M, N, K1, K2 in sdxl_attn_shapes:
+        configs.append(AttentionConfig(B, M, N, K1, K2, dtype))
+
+
+def get_attention_configs() -> list:
+    configs = []
+    llm_sweep(configs, "f16")
```

Comment:
Any plans for adding in fp8 attention as well? 😄

Reply:
If you add it, it should just work. I'm just not sure how to verify accuracy of fp8 attention, so it's not really easy to add it.